spring-integration-core/src/test/java/org/springframework/integration/transformer/ObjectToMapTransformerTests.java

```diff
@@ -154,6 +161,55 @@ public void testObjectToSpelMapTransformerWithCycle() {
 		transformer.transform(message);
 	}
 
+	@Test
+	public void testJacksonJSR310Support_PassInstantField_ReturnsMapWithOnlyOneEntryForInstantField() throws Exception {
+		/*
+		 * This test will fail if Jackson is not present in the classpath. (For JUnits Jackson will always be present)
+		 */
+		// arrange
+		Person person = new Person();
+		person.deathDate = Instant.now();
+
+		Employee employee = new Employee();
+		employee.setPerson(person);
+
+		// act
+		Map<String, Object> transformedMap = new ObjectToMapTransformer().transformPayload(employee);
+
+		// assert
+		assertEquals("If JSR310 support is enabled by calling findAndRegisterModules() on the Jackson mapper, " +
+						"Instant field should not be broken. Thus the count should exactly be 1 here.",
```

Comment:
The assert messages are not for explaining features.
Please, move it to the comment.
If you want to say here something that really should be a warn why the assert hasn't passed.